### PR TITLE
chore(deps)!: Bump github.com/ClickHouse/clickhouse-go from v1.5.4 to v2.18.0

### PR DESCRIPTION
**Pull request by srebhan**
## Summary

Version 1 of the ClickHouse SQL driver is no longer actively maintained. Therefore update to the new version 2.

**Note:** The DSN of v1 and v2 differs slightly, you might need to adapt your settings!

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

File: plugins/inputs/sql/sql.go [hunk 1]
```diff
@@ -413,6 +413,12 @@ func (s *SQL) ping() error {
 }
 
 func (s *SQL) prepareStatements() {
+	// We don't support prepare on ClickHouse as this requires a non-standard
+	// exec sequence on query later.
+	if s.driverName == "clickhouse" {
+		return
+	}
```
Comment (powersj) [inline plugins/inputs/sql/sql.go:420]: Did this work wtih the v1 library?

Comment (srebhan) [reply plugins/inputs/sql/sql.go:420]: I think so as the test did not complain. We can work on doing it (we already do have a special handling on the sql output plugin). The biggest issue is that `Query()` and `Exec()` will return two completely different data-types and we would need to implement this for only ClickHouse... Would you prefer we are doing this?

Comment (powersj) [reply plugins/inputs/sql/sql.go:420]: > Would you prefer we are doing this?

I'm not sure. From a let's not break anything that already works standpoint, then yes, but that seems quite involved. Are any of the other clickhouse libraries support this better out of the box?

**Review by Hipska — APPROVED** [review]

File: plugins/inputs/sql/sql.go (hunk 1 above)
Comment (powersj) [reply plugins/inputs/sql/sql.go:420]: Thanks - I agree on holding.

**Pull request closed without merging**